backend/src/federation/validateCommunities.ts

```diff
@@ -0,0 +1,74 @@
+import { Community as DbCommunity } from '@entity/Community'
+import { IsNull } from '@dbTools/typeorm'
+// eslint-disable-next-line camelcase
+import { requestGetPublicKey as v1_0_requestGetPublicKey } from './client/1_0/FederationClient'
+// eslint-disable-next-line camelcase
+import { requestGetPublicKey as v1_1_requestGetPublicKey } from './client/1_1/FederationClient'
+import { backendLogger as logger } from '@/server/logger'
+import { ApiVersionType } from './enum/apiVersionType'
+
+export async function startValidateCommunities(timerInterval: number): Promise<void> {
+  logger.info(
+    `Federation: startValidateCommunities loop with an interval of ${timerInterval} ms...`,
+  )
+  // better to use setTimeout twice than setInterval once -> see https://javascript.info/settimeout-setinterval
+  setTimeout(function run() {
+    validateCommunities()
```

Comment:
await to not trigger this twice if `validateCommunities` takes longer then `timerInterval` (you can probl. test with 1)

Reply:
that's exactly the solution to ensure a fix duration of timerInterval between the end of the previous invokation and the start of the next invokation independent how long the previous invokation needed.
the use of timerInterval will ensure a fix duration of timererInterval between the start of the previous invokation and the start of the next invokation. Here you're overlapping constellation could occur as you see here: https://javascript.info/settimeout-setinterval